File: leetcode/1561/solution.py
"""
https://leetcode.com/problems/maximum-number-of-coins-you-can-get/
1561. Maximum Number of Coins You Can Get
"""


class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        piles.sort()
        t = math.ceil(len(piles) / 3)
        return sum(piles[i - 1] for i in range(len(piles) - 1, t, -2))

File: leetcode/1561/solution.rs
/**
 * https://leetcode.com/problems/maximum-number-of-coins-you-can-get/
 * 1561. Maximum Number of Coins You Can Get
 */

impl Solution {
    pub fn max_coins(mut piles: Vec<i32>) -> i32 {
        piles.sort_unstable();
        let t = piles.len() / 3;
        piles.iter().skip(t).step_by(2).sum()
    }
}
